controllers/clusterdevfileregistrieslist_controller.go

```diff
@@ -68,10 +76,43 @@ func (r *ClusterDevfileRegistriesListReconciler) Reconcile(ctx context.Context,
 		return ctrl.Result{}, err
 	}
 
+	if clusterDevfileRegistriesList.Status.Conditions == nil || len(clusterDevfileRegistriesList.Status.Conditions) == 0 {
+		meta.SetStatusCondition(&clusterDevfileRegistriesList.Status.Conditions, metav1.Condition{
+			Type:    typeAvailableClusterDevfileRegistriesList,
+			Status:  metav1.ConditionUnknown,
+			Reason:  "Reconciling",
+			Message: "Starting reconciliation",
+		})
+		if err = r.Status().Update(ctx, clusterDevfileRegistriesList); err != nil {
+			log.Error(err, "Failed to update ClusterDevfileRegistriesList status")
+			return ctrl.Result{}, err
+		}
+
+		// re-fetch the Custom Resource after update the status
+		// so that we have the latest state of the resource on the cluster and we will avoid
+		// raise the issue "the object has been modified, please apply
+		// your changes to the latest version and try again" which would re-trigger the reconciliation
+		if err := r.Get(ctx, req.NamespacedName, clusterDevfileRegistriesList); err != nil {
```

Comment:
You shouldn't need to re-get the resource after updating the status, especially since the controller uses cached responses for the Kube API, it's not guaranteed that the resource returned here will even be the newest version. 

What scenarios were you seeing the update conflict in?

Reply:
I think it's related to this documented limitation for the list CRs: https://github.com/devfile/registry-operator/blob/main/REGISTRIES_LISTS.md#updating-the-cluster-or-devfile-registries-list

Reply:
Ah, right. I was wondering why there would be conflicts if only the operator was modifying the resource.

In that case, it might be better to skip fetching the resource here, and call `retryOnConflict` when an error does occur.

Reply:
Addressed in https://github.com/devfile/registry-operator/pull/49/commits/33593bdd7b4a2b25c7bc3f5feafa69109841c45f